src/protocols/LockNotify.cpp

```diff
@@ -0,0 +1,88 @@
+#include "LockNotify.hpp"
+
+CHyprlandLockNotification::CHyprlandLockNotification(SP<CHyprlandLockNotificationV1> resource_) : m_resource(resource_) {
+    if (!m_resource->resource())
```

Comment:
if statements missing LIKELY/UNLIKELY

(not all need those but this one for example is obviously UNLIKELY)